tests/tacacs/utils.py

```diff
@@ -66,24 +66,31 @@ def log_exist(ptfhost, sed_command):
     pytest_assert(exist, "Not found data: {} in tacplus server log".format(data))
 
 
-def get_auditd_config_reload_line_count(duthost):
+def get_auditd_config_reload_timestamp(duthost):
     res = duthost.shell("sudo journalctl -u auditd --boot --no-pager | grep 'audisp-tacplus re-initializing configuration'") # noqa E501
     logger.info("aaa config file timestamp {}".format(res["stdout_lines"]))
 
-    return len(res["stdout_lines"])
+    if len(res["stdout_lines"]) == 0:
+        return ""
 
+    return res["stdout_lines"][-1]
 
-def change_and_wait_aaa_config_update(duthost, command, last_line_count=None, timeout=10):
-    if not last_line_count:
-        last_line_count = get_auditd_config_reload_line_count(duthost)
+
+def change_and_wait_aaa_config_update(duthost, command, last_timestamp=None, timeout=10):
+    if not last_timestamp:
+        last_timestamp = get_auditd_config_reload_timestamp(duthost)
+
+    # Wait 2 seconds to make sure auditd config re-init time stamp not dupe with last timestamp:
+    #     https://github.com/sonic-net/sonic-mgmt/issues/16709
+    time.sleep(2)
```

Comment:
>2 [](http://example.com/codeflow?start=15&length=1)

Fix number of waiting second is not perfect. Could you set larger number of second, but periodically check, as long as the expected timestamp appear in journalctl output, stop the wait.

Reply:
Fixed
wait_until have a 1seconds sleep, add a command to trigger config reload when log does not exist.